File: Streaming/Python/reducer.py
import sys

lines = sys.stdin.readlines()
wordcount = {};
for line in lines:
    if line in wordcount:
	count = wordcount[line]
	count = count + 1
	wordcount[line] = count
    else:
	wordcount[line] = 1
for word in wordcount:
    sys.stdout.write(word.strip()+ "\t" + str(wordcount[word]) + "\n")

File: Streaming/php/reducer.php
<?php
$wordcount = array();
while(($myfile = fgets(STDIN)) != null)
{
	$file = trim($myfile);
	if(array_key_exists($file, $wordcount))
	{
		$count = $wordcount[$file];
		$count++;
		$wordcount[$file] = $count;
	}
	else
	{
		$wordcount[$file] = 1;
	}
}
foreach($wordcount as $key => $value)
{
	echo $key . "\t". $value . "\n";
}
?>
